db_stress_tool/db_stress_test_base.cc

```diff
@@ -2371,6 +2375,58 @@ Status StressTest::TestGetPropertiesOfAllTables() const {
   return db_->GetPropertiesOfAllTables(&props);
 }
 
+void StressTest::TestGetUpdatesSince(ThreadState* thread) {
+  const SequenceNumber since_seq =
+      thread->rand.Next64() % db_->GetLatestSequenceNumber();
+  std::unique_ptr<TransactionLogIterator> iter;
+
+  Status s = db_->GetUpdatesSince(since_seq, &iter);
+
+  if (!s.ok()) {
+    fprintf(stderr,
+            "Failed to call GetUpdatesSince(): %s with specified "
+            "seq num: %" PRIu64 "\n",
+            s.ToString().c_str(), since_seq);
+    thread->shared->SafeTerminate();
+  }
+
+  if (!iter->Valid()) {
+    return;
+  }
+
+  BatchResult res = iter->GetBatch();
+  SequenceNumber last_seqno = res.sequence;
+  iter->Next();
+
+  while (iter->Valid()) {
+    if (!iter->status().ok()) {
```

Comment:
GetUpdateSince() seems to expect WAL to contain updates with consecutive sequence numbers. It can fail here when there is a concurrent file ingestion: #10007. There may be other incompatibilities like writes with disable_wal can cause a hole in sequence numbers in WAL too.

Reply:
Thanks - need to look into more of this.

Reply:
Fixed and launching more tests to check

Reply:
Found more incompatibility run - need to fix them